grpc-java/grpc-servlet-websocket-jakarta/src/main/java/io/grpc/servlet/web/websocket/MultiplexedWebSocketServerStream.java

```diff
@@ -104,6 +177,14 @@ public void onMessage(ByteBuffer message) throws IOException {
 
         // if this is the first message on this websocket, it is the request headers
         if (stream == null) {
+            if (this.closed == ClosedState.CLOSED) {
+                // Not accepting more calls, client sent this before they saw that, be clear that this is closed
```

Comment:
this comment is wrong.

In this `if`, we know that the client is aware that they shouldnt open new streams but did it anyway, so we are closing them.

If closed == CLOSING, the client doesn't yet know, so like h2 GO_AWAY, we will permit them to start this new stream - they'll know very soon though, and will have a short window to wrap it up (and, in all likelihood, their session has already ended, so they may not be able to do what they want to do anyway, but that's an application-level concern, not a transport issue).